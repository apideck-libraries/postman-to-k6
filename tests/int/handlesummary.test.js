/* eslint-disable space-before-function-paren */

import convertFile from '../../lib/convert/file';

test('k6 handle-summary json', async () => {
  const options = {
    k6HandleSummary: {
      json: 'k6-handle-summary.json',
    },
  };
  const [main] = await convertFile('test/material/2/request.json', options);
  expect(main).toMatchSnapshot();
});
