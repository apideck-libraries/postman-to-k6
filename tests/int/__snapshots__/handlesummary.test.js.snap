// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`k6 handle-summary json 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export function handleSummary(data) {
  console.log("Preparing the end-of-test summary: ");
  return {
    "k6-handle-summary.json": JSON.stringify(data)
  };
}

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/"
  });
}
"
`;
